### PR TITLE
Add tests for body property handling in Request constructor

### PR DESCRIPTION
**Pull request by yutakahirano**
This change is for https://github.com/whatwg/fetch/pull/425 and https://github.com/whatwg/fetch/pull/458.

### CODE REVIEW

**Comment by jdm** [comment]
These new tests look good to me. I'm not going to merge them until someone else says that they're testing the cases we care about for the fetch PRs mentioned in the original comment.

**Comment by yutakahirano** [comment]
cc: @annevk

File: fetch/api/request/request-disturbed.html [hunk 1]
```diff
@@ -41,11 +41,30 @@
...
 
+      promise_test(() => {
+        const bodyRequest = new Request("", initValuesDict);
+        const originalBody = bodyRequest.body;
+        assert_false(bodyRequest.bodyUsed , "bodyUsed is false when request is not disturbed");
+        const requestFromRequest = new Request(bodyRequest, { body : "init body" });
+        assert_true(bodyRequest.bodyUsed , "bodyUsed is true when request is disturbed");
+        assert_equals(bodyRequest.body, originalBody, "body should not change");
+        assert_not_equals(originalBody, undefined, "body should not be undefined");
+        assert_not_equals(originalBody, null, "body should not be null");
```
Comment (annevk) [inline fetch/api/request/request-disturbed.html]: Can we add a test here about what it should be? Either instanceof/typeof or some such?

And perhaps we should also have a test where the initial request does not have a body?

File: fetch/api/request/request-disturbed.html [hunk 2]
```diff
@@ -41,11 +41,30 @@
...
       test(function() {
         var bodyRequest = new Request("", initValuesDict);
+        const originalBody = bodyRequest.body;
         assert_false(bodyRequest.bodyUsed , "bodyUsed is false when request is not disturbed");
         var requestFromRequest = new Request(bodyRequest);
         assert_true(bodyRequest.bodyUsed , "bodyUsed is true when request is disturbed");
+        assert_equals(bodyRequest.body, originalBody, "body should not change");
+        assert_not_equals(originalBody, undefined, "body should not be undefined");
+        assert_not_equals(originalBody, null, "body should not be null");
       }, "Input request used for creating new request became disturbed");
```
Comment (annevk) [inline fetch/api/request/request-disturbed.html]: This test is basically duplicated no as a promise_test, no? Maybe it's better to just remove it? Or am I missing something subtle?

Comment (annevk) [inline fetch/api/request/request-disturbed.html]: Oh wait, the difference is that the promise_test passes a body. Disregard the above comment please.

File: fetch/api/request/request-disturbed.html (hunk 1 above)
Comment (yutakahirano) [inline fetch/api/request/request-disturbed.html]: Oops, I found that I forgot to expose body property on Request. I'll make another fetch spec PR...

Comment (yutakahirano) [inline fetch/api/request/request-disturbed.html]: Anyway, I added some assertions.

Comment (annevk) [inline fetch/api/request/request-disturbed.html]: Maybe just add that to the existing PR? Then we can use these tests for both.

Comment (yutakahirano) [inline fetch/api/request/request-disturbed.html]: Ok, will do.

Comment (annevk) [inline fetch/api/request/request-disturbed.html]: Never mind, what you did is fine.

**Pull request merged**